flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecLookupJoin.java

```diff
@@ -246,18 +254,23 @@ protected Transformation<RowData> createJoinTransformation(
         RowType tableSourceRowType = FlinkTypeFactory.toLogicalRowType(temporalTable.getRowType());
         RowType resultRowType = (RowType) getOutputType();
         validateLookupKeyType(lookupKeys, inputRowType, tableSourceRowType);
+        boolean isAsyncEnabled = null != asyncLookupOptions;
+        ResultRetryStrategy retryStrategy =
+                retryOptions != null ? retryOptions.toRetryStrategy() : null;
 
         UserDefinedFunction lookupFunction =
                 LookupJoinUtil.getLookupFunction(
                         temporalTable,
                         lookupKeys.keySet(),
                         planner.getFlinkContext().getClassLoader(),
-                        joinHintSpec,
-                        upsertMaterialize);
+                        isAsyncEnabled,
+                        retryStrategy);
         UserDefinedFunctionHelper.prepareInstance(config, lookupFunction);
 
         boolean isLeftOuterJoin = joinType == FlinkJoinType.LEFT;
-        boolean isAsyncEnabled = lookupFunction instanceof AsyncTableFunction;
+        if (isAsyncEnabled) {
+            assert lookupFunction instanceof AsyncTableFunction;
```

Comment:
does this check need for online environment ?

Reply:
This never happens in runtime by design, but to prevent incorrectly changes in LookupJoinUtil#isAsyncLookup